WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/configuration/GetChildrenProductInfo.kt

```diff
@@ -0,0 +1,34 @@
+package com.woocommerce.android.ui.orders.creation.configuration
+
+import com.woocommerce.android.ui.products.GetBundledProducts
+import com.woocommerce.android.ui.products.ProductDetailRepository
+import com.woocommerce.android.ui.products.ProductType
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class GetChildrenProductInfo @Inject constructor(
+    private val productDetailRepository: ProductDetailRepository,
+    private val getBundledProducts: GetBundledProducts
+) {
+    operator fun invoke(productId: Long): Flow<Map<Long, ProductInfo>> {
+        val product = productDetailRepository.getProduct(productId) ?: return flowOf(emptyMap())
+        return when (product.productType) {
+            ProductType.BUNDLE -> {
+                getBundledProducts(productId).map { list ->
+                    val result = mutableMapOf<Long, ProductInfo>()
+                    list.onEach { bundledProduct ->
+                        result[bundledProduct.id] = ProductInfo(
+                            id = bundledProduct.id,
+                            title = bundledProduct.title,
+                            imageUrl = bundledProduct.imageUrl
+                        )
+                    }
+                    result
+                }
+            }
+            else -> flowOf(emptyMap())
+        }
```

Comment:
Refactoring suggestions: 

1. Instead of manually operating a new mutable map one item at a time, you can transform the entire list at once and then associate it by the id using the `associateBy` function. 
2. Since you're expecting only the `ProductType.BUNDLE` and rejecting everything else, you can use the takeIf on top of the getProduct call instead of using a `when`. 
3. It's possible to move everything else to an `emptyMap()` through a single Elvis operator, funneling the two failure scenarios to just one single point and the entire operation to a single pipe of transformations.

```suggestion
        return productDetailRepository.getProduct(productId)
            ?.takeIf { it.productType == ProductType.BUNDLE }
            ?.let { getBundledProducts(productId) }
            ?.map { products ->
                products.map { bundledProduct ->
                    ProductInfo(
                        id = bundledProduct.id,
                        title = bundledProduct.title,
                        imageUrl = bundledProduct.imageUrl
                    )
                }.associateBy { it.id }
            } ?: flowOf(emptyMap())
```